subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/ivyservice/resolveengine/artifact/ArtifactBackedResolvedVariant.java

```diff
@@ -60,7 +60,12 @@ public static ResolvedVariant create(@Nullable VariantResolveMetadata.Identifier
 
     private static Supplier<ResolvedArtifactSet> supplyResolvedArtifactSet(DisplayName displayName, AttributeContainerInternal attributes, CapabilitiesMetadata capabilities, Supplier<Collection<? extends ResolvableArtifact>> artifactsSupplier) {
         return () -> {
-            Collection<? extends ResolvableArtifact> artifacts = artifactsSupplier.get();
+            Collection<? extends ResolvableArtifact> artifacts;
+            try {
+                artifacts = artifactsSupplier.get();
+            } catch (Exception e) {
+                return EMPTY;
+            }
             if (artifacts.isEmpty()) {
```

Comment:
If there was an easy way to pass information about whether we are resolving leniently or strictly to the `ArtifactBackedResolvedVariant#create` method as it calls `supplyResolvedArtifacts` so that we can add this (or similar error handling) when this supplier's `get()` is called, this would be a straightforward change, I think.

Alternately, if we could change the `Supplier` to a `Function` and pass in the configuration doing the resolving when `get()/call()` is called, we could do the same thing.

As it is now, passing this information all the way into this `Supplier` when it is created seems like it would require sending a flag down a very long chain of resolution calls.